File: Old Java Code/Client/org/hardgate/familytree/IGIResult.java
/*
 * Created on 17-May-2005
 *
 */
package org.hardgate.familytree;

import java.net.MalformedURLException;
import java.net.URL;

import org.htmlparser.tags.LinkTag;

public class IGIResult {
    
	int searchType;
	String parish;
	String batch;
    String person;
    String gender;
    String spouse;
    String father;
    String mother;
    String birth;
    String christening;
    String death;
    String burial;
    String marriage;
    LinkTag link;
    
    public IGIResult(int searchType, ParishBatch pb, LinkTag link) {
        this.searchType = searchType;
        this.parish = pb.getParish();
        this.batch = pb.getBatch();
    	this.link = link;
        this.person = link.getLinkText();
        this.gender = "";
        this.spouse = "";
        this.mother = "";
        this.father = "";
        this.birth = "";
        this.christening = "";
        this.death = "";
        this.burial = "";   
        this.marriage = "";
    }
    
    public void updateValue(String field, String value) {
        if(field.indexOf("Birth") != -1)
            this.birth = value;
        else if(field.indexOf("Christening") != -1)
            this.christening = value;
        else if(field.indexOf("Death") != -1)
            this.death = value;
        else if(field.indexOf("Burial") != -1)
            this.burial = value;
        else if(field.indexOf("Father") != -1)
            this.father = value;
        else if(field.indexOf("Mother") != -1)
            this.mother = value;
        else if(field.indexOf("Spouse") != -1)
            this.spouse = value;
        else if(field.indexOf("Marriage") != -1)
            this.marriage = value;
    }
    
    public URL getURL() throws MalformedURLException {
        return new URL(link.getLink());
    }
    
    public String getMother() {
        // return only the XXXX bits between ">XXXX</a>
        return noLink(mother);
    }

    public String getPerson() {
        return person;
    }

    public String getSpouse() {
        // return only the XXXX bits between ">XXXX</a>
        return noLink(spouse);
    }

    public String getBirth() {
        return noNBSP(birth);
    }

    public String getFather() {
        // return only the XXXX bits between ">XXXX</a>
        return noLink(father);
    }

    public String getBurial() {
        return noNBSP(burial);
    }

    public String getChristening() {
        return noNBSP(christening);
    }

    public String getDeath() {
        return noNBSP(death);
    }

    public String getGender() {
        return gender;
    }

    public void setGender(String gender) {
        this.gender = gender;
    }
    
    public String getMarriage() {
        return noNBSP(marriage);
    }

    private String noNBSP(String input) {
        String temp = input;
        String output = "";
        while (temp.indexOf("&nbsp;") !=-1) {
            int pos = temp.indexOf("&nbsp;");
            if(pos > 0)
                output = output + temp.substring(0,pos);
            temp = temp.substring(pos+6);
        }
        if(temp.length() > 0)
            output = output + temp;
        return output;
    }

    private String noLink(String input) {
        String output = input;
        if (input.indexOf(">") !=-1) {
            int start = input.indexOf(">") + 1;
            int end = input.indexOf("</a>") == -1 ? 
                    input.length() : input.indexOf("</a>");
            output = input.substring(start,end);
        }
        return output;
    }

	public int getSearchType() {
		return searchType;
	}

	public String getBatch() {
		return batch;
	}

	public void setBatch(String batch) {
		this.batch = batch;
	}

	public String getParish() {
		return parish;
	}

	public void setParish(String parish) {
		this.parish = parish;
	}
}

File: FTAnalyser/Searches/IGIResult.cs
using System;
using System.IO;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Net;

namespace FTAnalyzer
{
    public class IGIResult {
        
	    int searchType;
	    Parish parish;
	    String batch;
        String person;
        String gender;
        String spouse;
        String father;
        String mother;
        String birth;
        String christening;
        String death;
        String burial;
        String marriage;
        String link;
        
        public IGIResult(int searchType, ParishBatch pb, string link) {
            this.searchType = searchType;
            this.parish = pb.Parish;
            this.batch = pb.Batch;
    	    this.link = link;
            this.person = link;
            this.gender = "";
            this.spouse = "";
            this.mother = "";
            this.father = "";
            this.birth = "";
            this.christening = "";
            this.death = "";
            this.burial = "";   
            this.marriage = "";
        }
        
        public void updateValue(String field, String value) {
            if(field.IndexOf("Birth") != -1)
                this.birth = value;
            else if(field.IndexOf("Christening") != -1)
                this.christening = value;
            else if(field.IndexOf("Death") != -1)
                this.death = value;
            else if(field.IndexOf("Burial") != -1)
                this.burial = value;
            else if(field.IndexOf("Father") != -1)
                this.father = value;
            else if(field.IndexOf("Mother") != -1)
                this.mother = value;
            else if(field.IndexOf("Spouse") != -1)
                this.spouse = value;
            else if(field.IndexOf("Marriage") != -1)
                this.marriage = value;
        }
        
        public HttpWebRequest URL {
            get { return (HttpWebRequest)WebRequest.Create(link); }
        }
        
        public String Mother {
            // return only the XXXX bits between ">XXXX</a>
            get { return noLink(mother); }
        }

        public String Person {
            get { return person; }
        }

        public String Spouse {
            // return only the XXXX bits between ">XXXX</a>
            get { return noLink(spouse); }
        }

        public String Birth {
            get { return noNBSP(birth); }
        }

        public String Father {
            // return only the XXXX bits between ">XXXX</a>
            get { return noLink(father); }
        }

        public String Burial {
            get { return noNBSP(burial); }
        }

        public String Christening {
            get { return noNBSP(christening); }
        }

        public String Death {
            get { return noNBSP(death); }
        }

        public String Gender {
            get { return gender; }
            set { this.gender = value; }
        }
     
        public String Marriage {
            get { return noNBSP(marriage); }
        }

        public int SearchType
        {
            get { return searchType; }
        }

        public String Batch
        {
            get { return batch; }
            set { this.batch = value; }
        }

        public Parish Parish
        {
            get { return parish; }
            set { this.parish = value; }
        }

        private String noNBSP(String input)
        {
            String temp = input;
            String output = "";
            while (temp.IndexOf("&nbsp;") !=-1) {
                int pos = temp.IndexOf("&nbsp;");
                if(pos > 0)
                    output = output + temp.Substring(0,pos);
                temp = temp.Substring(pos+6);
            }
            if(temp.Length > 0)
                output = output + temp;
            return output;
        }

        private String noLink(String input) {
            String output = input;
            if (input.IndexOf(">") !=-1) {
                int start = input.IndexOf(">") + 1;
                int end = input.IndexOf("</a>") == -1 ? input.Length : input.IndexOf("</a>");
                output = input.Substring(start,end-start);
            }
            return output;
        }
    }
}